torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,746 @@
+
+        self.grus = nn.ModuleList(
+            [
+                ConvGRU(input_size=gru_input_dims[i], hidden_size=hidden_dims[i], kernel_size=3, padding=1)
+                for i in range(len(hidden_dims))
+            ]
+        )
+
+        self.hidden_dims = hidden_dims
+
+    def _downsample2x(self, x):
+        return F.avg_pool2d(x, 3, stride=2, padding=1)
+
+    def _upsample2x(self, x):
+        _, _, h, w = x.shape
+        new_h, new_w = h * 2, w * 2
+        return F.interpolate(x, size=(new_h, new_w), mode="bilinear", align_corners=True)
+
+    def forward(
+        self,
+        hidden_states: List[Tensor],
+        contexts: List[Tuple[Tensor, Tensor, Tensor]],
+        corr_features: Tensor,
+        depth: Tensor,
+        level_processed: List[int] = [1, 1, 1],
+    ) -> List[Tensor]:
+        for i in range(len(self.grus) - 1, -1, -1):
+            if level_processed[i]:
+                # X is concatination of downsampled hidden_dim (or motion_features if no bigger dim) with
+                # upsampled hidden_dim (or nothing if not exist)
+                features = (
+                    self._downsample2x(hidden_states[i - 1]) if i > 0 else self.motion_encoder(depth, corr_features)
+                )
+                if i < len(self.grus) - 1:
+                    features = torch.cat([features, self._upsample2x(hidden_states[i + 1])], dim=1)
+
+                hidden_states[i] = self.grus[i](hidden_states[i], contexts[i], features)
+
+                # NOTE: For slow-fast gru, we dont always want to calculate delta depth for every call on UpdateBlock
+                # Hence we move the delta depth calculation to the RAFT-Stereo main forward
+
+        return hidden_states
+
+
+class MaskPredictor(raft.MaskPredictor):
+    """Mask predictor to be used when upsampling the predicted depth."""
+
+    # We add out_channels compared to raft.MaskPredictor
+    def __init__(self, *, in_channels: int, hidden_size: int, out_channels: int, multiplier: float = 0.25):
+        super(raft.MaskPredictor, self).__init__()
+        self.convrelu = Conv2dNormActivation(in_channels, hidden_size, norm_layer=None, kernel_size=3)
+        self.conv = nn.Conv2d(hidden_size, out_channels, kernel_size=1, padding=0)
+        self.multiplier = multiplier
+
+
+class CorrBlock1d(nn.Module):
+    """The row-wise correlation block.
+
+    Creates a row-wise correlation pyramid with ``num_levels`` levels from the outputs of the feature encoder,
+    and then indexes from this pyramid to create correlation features.
+    The "indexing" of a given centroid pixel x' is done by concatenating its surrounding row neighbours
+    within radius
+    """
+
+    def __init__(self, *, num_levels: int = 4, radius: int = 4):
+        super().__init__()
+        self.num_levels = num_levels
+        self.radius = radius
+
+        self.corr_pyramid: List[Tensor] = [torch.tensor(0)]  # useless, but torchscript is otherwise confused :')
```

Comment:
Show us the message you get if you oit the torch.tensor here. Let's see if we can fix JIT without the workaround.

Reply:
This class is actually heavily modified from the raft: https://github.com/pytorch/vision/blob/main/torchvision/models/optical_flow/raft.py#L328

This part comes from there. Let me try it out.

Reply:
Actually thanks to this, I realized the model is not yet jit scriptable. After make change to make sure the model is jit scriptable, when I remove the `torch.tensor(0)` I will got the following error message:
```
RuntimeError: 
Tried to set nonexistent attribute: corr_pyramid. Did you forget to initialize it in __init__()?:
  File "/Users/yosuamichael/manual_build_lib/pytorch/vision/torchvision/prototype/models/depth/stereo/raft_stereo.py", line 251
        batch_size, h, w, num_channels, _ = corr_volume.shape
        corr_volume = corr_volume.reshape(batch_size * h * w, num_channels, 1, w)
        self.corr_pyramid = [corr_volume]
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        for _ in range(self.num_levels - 1):
            corr_volume = F.avg_pool2d(corr_volume, kernel_size=(1, 2), stride=(1, 2))
'CorrBlock1d.build_pyramid' is being compiled since it was called from 'RaftStereo.forward'
  File "/Users/yosuamichael/manual_build_lib/pytorch/vision/torchvision/prototype/models/depth/stereo/raft_stereo.py", line 383
            raise ValueError(f"The feature encoder should downsample H and W by {self.base_downsampling_ratio}")
    
        self.corr_block.build_pyramid(fmap1, fmap2)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE

```

Reply:
After separating the corr_block into 2 modules, and putting the `corr_pyramid` in the `RaftStereo` forward, this issue is not relevant anymore.